### ISSUE TITLE
Attempt to run an application that does not exist creates zombie processes on Linux

### ISSUE THREAD

**Issue by arybczak** [haskell/process#18]
The following program:

```haskell
{-# OPTIONS_GHC -Wall #-}
import Control.Concurrent
import Control.Exception
import Control.Monad
import System.Process

main :: IO ()
main = forever $ do
  putStrLn =<< readProcess "does_not_exist" [] [] `catch` printEx
  threadDelay 1000000
  where
    printEx :: SomeException -> IO String
    printEx = return . show
```

Produces one zombie process per second (assuming that application 'does_not_exist' does not actually exist). The culprit seems to be the fact that exception occurs in `createProcess_`, so the bracket in `withCreateProcess_` doesn't help because `createProcess_` is the "acquire resource" part.

This is with GHC 7.8.4 and process-1.2.0.0.

**Comment by hvr** [comment]
sounds related to #14

**Comment by arybczak** [comment]
Indeed, it's fixed in 1.2.2.0, thanks.
